### ISSUE TITLE
DataMatrix is missing support for damaged  symbols

### ISSUE THREAD

**Issue by aiden-jiang** [nu-book/zxing-cpp#234]
![X12damaged ](https://user-images.githubusercontent.com/85624698/121453460-c8424200-c9d3-11eb-8137-cde9232c5b82.png)
the black/white borders  on the right is broken,I using the zxing-cpp 1.2.0,the decode result is failed

**Comment by axxel** [comment]
Yes, the current code does not support this type of damage to the datamatrix symbol. You are welcome to suggest improvements to the code that would support this. I won't spent time on this as there are more important aspects of the library to be improved.

**Comment by aiden-jiang** [comment]
my idea is when  can't find the  black/white borders  on the right，don't return failure，continue to look for the top border，In later code，using the length-width ratio to get dimR

**Comment by axxel** [comment]
I'd be happy to look at a PR from you. As I said, I don't have the time to work on that myself.
